backend/dataall/modules/datapipelines/api/resolvers.py

```diff
@@ -205,28 +187,30 @@ def get_job_runs(context, source: DataPipeline, **kwargs):
     if not source:
         return None
     with context.engine.scoped_session() as session:
-        task = models.Task(targetUri=source.DataPipelineUri, action='glue.job.runs')
-        session.add(task)
+        try:
+            response = DataPipelineService.get_job_runs(
+                session=session,
+                datapipeline_uri=source.DataPipelineUri
+            )
+        except Exception as e:
+            log.error(f"Failed to execute task due to: {e}")
 
-    response = Worker.process(
-        engine=context.engine, task_ids=[task.taskUri], save_response=False
-    )[0]
-    return response['response']
+        return response[0]['response']
 
 
 def get_pipeline_executions(context: Context, source: DataPipeline, **kwargs):
     if not source:
         return None
     with context.engine.scoped_session() as session:
-        task = models.Task(
-            targetUri=source.DataPipelineUri, action='datapipeline.pipeline.executions'
-        )
-        session.add(task)
+        try:
+            response = DataPipelineService.get_pipeline_execution(
+                session=session,
+                datapipeline_uri=source.DataPipelineUri
+            )
+        except Exception as e:
+            log.error(f"Failed to execute task due to: {e}")
```

Comment:
Minor: we can improve the error message. e.g.` Failed to find pipeline execution for {source.DataPipelineUri}. Error {e}`